projects/core/src/config/config-initializer/config-initializer.ts

```diff
@@ -0,0 +1,8 @@
+import { InjectionToken } from '@angular/core';
+
+export const CONFIG_INITIALIZER = new InjectionToken('ConfigInitializer');
+
+export interface ConfigInitializer {
+  scopes: string[];
+  configFactory: () => {};
```

Comment:
What about simply `factory`?
What about type `() => Promise<any> | void` - for self-documenting?

Reply:
Agree with `Promise<any>` 👍 
Any specific reason for 'void'? 
Shouldn't config initializer always provide config chunk?

Reply:
2. `configFactory` is a little bit more obvious to me (a factory that should create config)